### PR TITLE
`Assessment`: Focus inline feedback editor when reviewing programming submissions

### LINKED ISSUE #6675: `Programming Exercises`, `Manual Assessment`: Cursor Focus for Tutor Comment Text Box
### Is your feature request related to a problem?

When creating a tutor comment in the assessment view by clicking at the `+` on the left of any line number, the mouse is not inside the feedback text box. This can be very frustrating, especially when the tutor expects the autofocus.

### Describe the solution you'd like

It would be nice if the focus was already inside the text field when the overlay is displayed, so that one could start typing immediately.

### Describe alternatives you've considered

There aren't really any alternatives, so none.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by milljoniaer**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/guidelines/development-process/#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

Resolves #6675 
Once a tutor inserts inline feedback inside programming exercise manual assessments, you want to start writing immediately. Therefore the textbox should be focused in edit mode, which was not the case.

### Description
There are two cases in which we want the textbox to be focused: 
- *initial open*: the angular components for each line are created at page load and get visible through an added class name (this is part of the ace-code-editor functionality). Therefore, we can not solve this inside the angular component but focus the textbox when the class name is added externally.
- *enter edit mode on existing feedback*: we can do this inside the angular component by reacting to the `editFeedback` "event"

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Students
- 1 Programming Exercise with Manual assessment

1. Log in to Artemis
2. Navigate to Course Administration > "your-exercise" > Exercise Dashboard > Start Assessment
3. Add new inline feedback and verify that the cursor is already inside the feedback textbox
4. Save the inline feedback
5. Click the `Edit` button and verify that the cursor is again inside the feedback textbox

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots

https://github.com/ls1intum/Artemis/assets/78978542/319079c6-b6ca-486c-8c93-c6d217a1a733

### CODE REVIEW

File: src/main/webapp/app/exercises/programming/assess/code-editor-tutor-assessment-inline-feedback.component.ts [hunk 1]
```diff
@@ -95,12 +96,13 @@ export class CodeEditorTutorAssessmentInlineFeedbackComponent {
     }
 
     /**
-     * Checks if component is in view mode
+     * Checks if component is in view mode and focus feedback text area
```
Comment (jakubriegel) [inline src/main/webapp/app/exercises/programming/assess/code-editor-tutor-assessment-inline-feedback.component.ts:99]: Small typo: _...and focus**es** feedback..._

**Review by jakubriegel — APPROVED** [review]
Code LGTM 👍🏻 
Issue verified and fix tested on ts11 ✅

**Comment by undernagruzez** [comment]
Tested on ts6, works as expected

**Review by chrisknedl — APPROVED** [review]
Code LGTM. Thanks for taking the time to implement this feature!

**Review by undernagruzez — APPROVED** [review]
lgtm

**Review by SabrinaGlatz — APPROVED** [review]
tested on ts5, works as expected

**Pull request merged**